src/status_im/activity_center/core.cljs

```diff
@@ -0,0 +1,69 @@
+(ns status-im.activity-center.core
+  (:require [re-frame.core :as re-frame]
+            [status-im.data-store.activities :as data-store.activities]
+            [status-im.ethereum.json-rpc :as json-rpc]
+            [status-im.utils.fx :as fx]
+            [taoensso.timbre :as log]))
+
+(def notifications-per-page
+  20)
+
+(def start-or-end-cursor
+  "")
+
+(defn notifications-group->rpc-method
+  [notifications-group]
+  (if (= notifications-group :notifications-read)
+    "wakuext_readActivityCenterNotifications"
+    "wakuext_unreadActivityCenterNotifications"))
+
+(defn notifications-read-status->group
+  [status-filter]
+  (if (= status-filter :read)
+    :notifications-read
+    :notifications-unread))
+
+(fx/defn notifications-fetch
+  [{:keys [db]} cursor notifications-group]
+  (when-not (get-in db [:activity-center notifications-group :loading?])
+    {:db             (assoc-in db [:activity-center notifications-group :loading?] true)
+     ::json-rpc/call [{:method     (notifications-group->rpc-method notifications-group)
+                       :params     [cursor notifications-per-page]
+                       :on-success #(re-frame/dispatch [:activity-center/notifications:fetch:success notifications-group %])
+                       :on-error   #(re-frame/dispatch [:activity-center/notifications:fetch:error notifications-group %])}]}))
+
+(fx/defn notifications-fetch-first-page
+  {:events [:activity-center/notifications:fetch-first-page]}
+  [{:keys [db] :as cofx} {:keys [status-filter] :or {status-filter :unread}}]
+  (let [notifications-group (notifications-read-status->group status-filter)]
+    (fx/merge cofx
+              {:db (-> db
+                       (assoc-in [:activity-center :current-status-filter] status-filter)
+                       (update-in [:activity-center notifications-group] dissoc :loading?)
+                       (update-in [:activity-center notifications-group] dissoc :data))}
+              (notifications-fetch start-or-end-cursor notifications-group))))
+
+(fx/defn notifications-fetch-next-page
+  {:events [:activity-center/notifications:fetch-next-page]}
+  [{:keys [db] :as cofx}]
+  (let [status-filter       (get-in db [:activity-center :current-status-filter])
+        notifications-group (notifications-read-status->group status-filter)
+        {:keys [cursor]}    (get-in db [:activity-center notifications-group])]
+    (when-not (= cursor start-or-end-cursor)
+      (notifications-fetch cofx cursor notifications-group))))
+
+(fx/defn notifications-fetch-success
+  {:events [:activity-center/notifications:fetch:success]}
```

Comment:
mgm im not sure we use `:` in events names, better just use `-` , `:activity-center/notifications-fetch-success`

Reply:
We don't, the idea of separating success with a `:` is new for sure (although common in other re-frame codebases). I'll change all events to use to `-` ;)

On the other hand, event names in our codebase are not very consistent, so we could eventually refactor them to a consistent naming scheme. For example, I've seen fully qualified event keywords, namespaced event keywords, events without namespaces, events starting with verbs, events starting with nouns, and so on.

I think there's no right or wrong, but we should be consistent. Perhaps a Github issue about this matter is in order?

Reply:
yes you are right, there is no right or wrong, definitely we should discuss and refactor, thank you

Reply:
Renaming events resolved by https://github.com/status-im/status-mobile/pull/13987/commits/ec33a92c64a2e257f4df57d269c5c8e39831ff44

Reply:
FYI: I created the issue [Lack of consistency on how re-frame events are named](https://github.com/status-im/status-mobile/issues/13996)